src/plugins/condition/components/inspector/StylesView.vue

```diff
@@ -0,0 +1,600 @@
+            this.conditionalStyles = [];
+            let domainObjectStyles =  (this.domainObject.configuration && this.domainObject.configuration.objectStyles) || {};
+            if (this.domainObjectsById) {
+                const domainObjects = Object.values(this.domainObjectsById);
+                domainObjects.forEach(domainObject => {
+                    let objectStyles =  (domainObject.configuration && domainObject.configuration.objectStyles) || {};
+                    this.removeConditionalStyles(objectStyles);
+                    if (_.isEmpty(objectStyles)) {
+                        objectStyles = undefined;
+                    }
+                    this.persist(domainObject, objectStyles);
+                });
+            }
+            if (this.items.length) {
+                this.items.forEach((item) => {
+                    const itemId = item.id;
+                    this.removeConditionalStyles(domainObjectStyles, itemId);
+                    if (_.isEmpty(domainObjectStyles[itemId])) {
+                        delete domainObjectStyles[itemId];
+                    }
+                });
+            }  else {
+                this.removeConditionalStyles(domainObjectStyles);
+            }
+            if (_.isEmpty(domainObjectStyles)) {
+                domainObjectStyles = undefined;
+            }
+            this.persist(this.domainObject, domainObjectStyles);
+
+            if (this.stopProvidingTelemetry) {
+                this.stopProvidingTelemetry();
+            }
+        },
+        removeConditionalStyles(domainObjectStyles, itemId) {
+            if (itemId && domainObjectStyles[itemId]) {
+                domainObjectStyles[itemId].conditionSetIdentifier = undefined;
+                delete domainObjectStyles[itemId].conditionSetIdentifier;
+                domainObjectStyles[itemId].selectedConditionId = undefined;
+                domainObjectStyles[itemId].defaultConditionId = undefined;
+                domainObjectStyles[itemId].styles = undefined;
+                delete domainObjectStyles[itemId].styles;
+            } else {
+                domainObjectStyles.conditionSetIdentifier = undefined;
+                delete domainObjectStyles.conditionSetIdentifier;
+                domainObjectStyles.selectedConditionId = undefined;
+                domainObjectStyles.defaultConditionId = undefined;
+                domainObjectStyles.styles = undefined;
+                delete domainObjectStyles.styles;
+            }
+        },
+        updateStaticStyle(staticStyle, property) {
+            //update the static style for each of the layoutItems as well as each sub object item
+            this.staticStyle = staticStyle;
+            this.removeConditionSet();
+            this.getAndPersistStyles(property);
+        },
+        updateConditionalStyle(conditionStyle, property) {
+            let found = this.findStyleByConditionId(conditionStyle.conditionId);
```

Comment:
can we use a more descriptive variable name, eg foundStyle ?

Reply:
Done